linkerd/app/src/dst/default_resolve.rs

```diff
@@ -24,12 +24,17 @@ where
     S::Endpoint: Default + Send + 'static,
     S::Resolution: Send + 'static,
     S::Future: Send + 'static,
-    stream::Once<future::Ready<Result<Update<S::Endpoint>, S::Error>>>:
-        stream::TryStream<Ok = Update<S::Endpoint>, Error = S::Error>,
+    stream::Chain<
+        stream::Once<future::Ready<Result<Update<S::Endpoint>, Error>>>,
+        stream::Pending<Result<Update<S::Endpoint>, Error>>,
+    >: stream::TryStream<Ok = Update<S::Endpoint>, Error = S::Error>,
 {
     type Response = future::Either<
         S::Resolution,
-        stream::Once<future::Ready<Result<Update<S::Endpoint>, Error>>>,
+        stream::Chain<
+            stream::Once<future::Ready<Result<Update<S::Endpoint>, Error>>>,
+            stream::Pending<Result<Update<S::Endpoint>, Error>>,
+        >,
```

Comment:
yes, i know this part is kinda terrible, sorry. we could box these if you prefer...